pkg/resources/resource_source_postgres.go

```diff
@@ -85,6 +77,90 @@ func SourcePostgres() *schema.Resource {
 	}
 }
 
+func sourcePostgresRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	i := d.Id()
+
+	metaDb, region, err := utils.GetDBClientFromMeta(meta, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	s, err := materialize.ScanSource(metaDb, utils.ExtractId(i))
+	if err == sql.ErrNoRows {
+		d.SetId("")
+		return nil
+	} else if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(utils.TransformIdWithRegion(string(region), i))
+
+	if err := d.Set("name", s.SourceName.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("schema_name", s.SchemaName.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("database_name", s.DatabaseName.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("size", s.Size.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("cluster_name", s.ClusterName.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("ownership_role", s.OwnerName.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	b := materialize.Source{SourceName: s.SourceName.String, SchemaName: s.SchemaName.String, DatabaseName: s.DatabaseName.String}
+	if err := d.Set("qualified_sql_name", b.QualifiedName()); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("comment", s.Comment.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	deps, err := materialize.ListDependencies(metaDb, utils.ExtractId(i), "source")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// TODO: We need to use a query that retrieves name and alias
+	// Tables
+	// tMaps := []interface{}{}
+	// for _, dep := range deps {
+	// 	tMap := map[string]interface{}{}
+	// 	tMap["name"] = dep.ObjectName.String
+	// 	tMap["alias"] = dep.ObjectName.String
+	// 	tMaps = append(tMaps, tMap)
+	// }
+	// if err := d.Set("table", tMaps); err != nil {
+	// 	return diag.FromErr(err)
+	// }
+
+	// Subsources
+	sMaps := []interface{}{}
+	for _, dep := range deps {
+		sMap := map[string]interface{}{}
+		sMap["name"] = dep.ObjectName.String
+		sMap["schema_name"] = dep.SchemaName.String
+		sMap["database_name"] = dep.DatabaseName.String
+		sMaps = append(sMaps, sMap)
+	}
+	if err := d.Set("subsource", sMaps); err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
We need to resolve the TODO here to actually fix #421, right? And for that I think we need to add a new table to Materialize like `mz_postgres_sources` that exposes the table name/alias mapping.

Reply:
cc @sploiselle

Reply:
It will work without the TODO fix, but on the first run it will throw: `ERROR: another subsource already refers to postgres.public.table3` if the subsource already exists as Terraform will not be aware of it and will try to create it anyway. But after that first run the state will be updated and it will work as expected, so there should be no disruption on the Materialize side, besides the annoying error that the user will get. Of course it will be much cleaner to have that TODO fixed as well.

Reply:
> It will work without the TODO fix, but on the first run it will throw: `ERROR: another subsource already refers to postgres.public.table3` if the subsource already exists as Terraform will not be aware of it and will try to create it anyway. But after that first run the state will be updated and it will work as expected, so there should be no disruption on the Materialize side, besides the annoying error that the user will get. Of course it will be much cleaner to have that TODO fixed as well.

Wait, that seems problematic in a different way. If adding the new table fails, why is Terraform updating the state? I would expect `terraform apply` to refrain from updating the state with the new table until it sees `ALTER SOURCE ... ADD SUBSOURCE` succeed? 

Another possible hazard: what if I manually drop a subsource out from under Terraform? I'd expect the next `terraform refresh && terraform apply` to recreate that subsource, but I think Terraform will never realize that that subsource has gone missing?

Anyway, sorry! I'm having trouble reasoning about the correctness of this change without being able to update the `tables` property! There's also something I don't understand about how the Terraform state is getting updated that seems spooky.

Adding a table to Materialize that exposes the mapping between PostgreSQL subsource table names/aliases is something that's pretty straightforward (could do in a week or two), and I'm wondering if we ought to do it before merging this PR, to make it easier to reason about the correctness here, and to not have to give users any caveats about the behavior of adding/removing subsources using Terraform after they do the migration that this PR requires.

Reply:
Indeed, it will be a much cleaner solution to have the table with the mapping between subsource table names.

Speaking of this, we should do the same for the MySQL source as well as I believe that we will be in the same situation there as well, cc @rjobanp